orchestrator/api/workspace_files.py

```
@@ -0,0 +1,221 @@
+    """
+    if "\x00" in relative_path:
+        raise PathSecurityError("Null byte in path")
+
+    if relative_path.startswith("/"):
+        raise PathSecurityError("Absolute path not allowed")
+
+    resolved = (workspace_root / relative_path).resolve()
+    base_resolved = workspace_root.resolve()
+
+    if not str(resolved).startswith(str(base_resolved)):
+        raise PathSecurityError("Path traversal blocked")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)